src/components/Select/hooks.js

```diff
@@ -0,0 +1,211 @@
+import { useEffect, useState } from "react";
+import { useClickOutside } from "hooks";
+import { blankValue, findExact } from "./helpers";
+import { useTransition } from "react-spring";
+
+export function useSelect(
+  value,
+  disabled,
+  options,
+  getOptionLabel,
+  getOptionValue,
+  autoFocus,
+  onChange,
+  inputValue,
+  onInputChange,
+  searchable = true,
+  error = false // Could not pass this props, since showError is created after SelectWrapper
+) {
+  const [_options, setOptions] = useState([]);
+  const [menuOpened, setMenuOpened] = useState(false);
+  const [focusedOptionIndex, setFocusedOptionIndex] = useState(0);
+
+  useEffect(() => {
+    if (disabled) {
+      setMenuOpened(false);
+      if (searchable && onInputChange && inputValue) onInputChange("");
+      return;
+    }
+    if (autoFocus) setMenuOpened(true);
+  }, [disabled, autoFocus, searchable, inputValue, onInputChange]);
+
+  useEffect(() => {
+    if (searchable && inputValue && !error) setMenuOpened(options.length > 0);
+  }, [searchable, inputValue, options, error]);
+
+  const resetMenu = (focusedIndex = 0) => {
+    setFocusedOptionIndex(focusedIndex);
+    setMenuOpened(false);
+    if (searchable && onInputChange && inputValue) onInputChange("");
+  };
+
+  const removeSelectedOptionFilter = (option) =>
+    value.filter(
+      (selectedOption) =>
+        getOptionLabel(selectedOption) !== getOptionLabel(option)
+    );
+
+  const setOption = Array.isArray(value)
+    ? (option, knownIndex) => {
+        if (disabled) return;
+        const index =
+          knownIndex ||
+          findExact(_options, getOptionLabel, getOptionValue, option);
+        let newSelectedOptions = [];
+        if (
+          value.find(
+            (selectedOption) =>
+              getOptionLabel(selectedOption) === getOptionLabel(option)
+          )
+        )
+          newSelectedOptions = removeSelectedOptionFilter(option);
+        else if (!value.length) {
+          newSelectedOptions = [option];
+        } else {
+          newSelectedOptions = [...value, option];
+        }
+        setFocusedOptionIndex(index);
+        if (searchable && onInputChange && inputValue) onInputChange("");
+        onChange(newSelectedOptions);
+      }
+    : (option, knownIndex) => {
+        const index =
+          knownIndex ||
+          findExact(_options, getOptionLabel, getOptionValue, option);
+        resetMenu(index);
+        onChange(option);
+      };
+
+  const [containerRef] = useClickOutside(resetMenu);
+
+  const selectOption = () => {
+    if (!menuOpened) return;
+    const optionIndex = focusedOptionIndex;
+    const optionByIndex = _options[optionIndex];
+    if (_options[optionIndex].onClick !== undefined) {
+      _options[optionIndex].onClick();
```

Comment:
are `null` and `false` values accepted? It seems you are checking for the existence of the `onClick` method...

can you use this instead?
```suggestion
    const optionByIndex = _options[optionIndex] || {};
    if (optionByIndex.onClick) {
       optionByIndex.onClick()
```